pkg/controller/pagerdutyintegration/pagerdutyintegration_controller.go

```diff
@@ -184,10 +171,21 @@ func (r *ReconcilePagerDutyIntegration) Reconcile(request reconcile.Request) (re
 		return r.requeueOnErr(err)
 	}
 
+	pdApiKey, err := utils.LoadSecretData(
+		r.client,
+		pdi.Spec.PagerdutyApiKeySecretRef.Name,
+		pdi.Spec.PagerdutyApiKeySecretRef.Namespace,
+		config.PagerDutyAPISecretKey,
+	)
+	if err != nil {
+		return r.requeueOnErr(err)
```

Comment:
I worry this will be a hot-loop of requeue.  I don't know if the sdk supports an exponential backoff but it should be used here.  Alternately we do not requeue this.  Reconcile will run periodically and we'll get a retry then.  Note the operator wouldn't have loaded in the prior implementation as the `Add` function would have raised an error.  We shouldn't do that now but we should be resilient to failures and strive for minimal impact on misconfiguration.

Additionally we should collect and export metrics on this failure so we can alert on it.  Misconfigured PDI need to be raised to on-call.

* Metric: pagerdutyinstance_secret_loaded
* Labels:
   * name = the PDI name
* Value:
   * 0 = unable to load referenced secret
   * 1 = able to load referenced secret

Then alert when metric value is `0` for some period of time, say `15m`.

To date we have put SRE alerts into openshiftmanaged-cluster-config.  But it doesn't have to be there.  I'd prefer we include this in the repo in `manifests` and bundle it into `hack/olm-registry/olm-artifacts-template.yaml` for deployment along with the operator.

Reply:
How about a requeue every 10 minutes on failure here?

Reply:
I think the reconcile loop is run for all resources every 10 minutes already, but I don't have a reference for that.  Having it requeue on some delay would be OK, though we won't know about it w/o metrics and alerts.

Reply:
Since there's no `SyncPeriod` option passed to the manager [here in main.go](https://github.com/openshift/pagerduty-operator/blob/master/cmd/manager/main.go#L107), then [the default is 10 hours](https://godoc.org/sigs.k8s.io/controller-runtime/pkg/manager#Options). I think if we want to set a `RequeueAfter` flag on the reconcile result, then we may need to swallow the error. That may be fine if we want to just log it and have a metric that we can alert on.